Mathlib/CategoryTheory/Arrow.lean

```diff
@@ -0,0 +1,336 @@
+
+The category of arrows, with morphisms commutative squares.
+We set this up as a specialization of the comma category `Comma L R`,
+where `L` and `R` are both the identity functor.
+
+## Tags
+
+comma, arrow
+-/
+
+
+namespace CategoryTheory
+
+universe v u
+
+-- morphism levels before object levels. See note [CategoryTheory universes].
+variable {T : Type u} [Category.{v} T]
+
+section
+
+variable (T)
+
+/-- The arrow category of `T` has as objects all morphisms in `T` and as morphisms commutative
+     squares in `T`. -/
+def Arrow :=
+  Comma.{v, v, v} (𝟭 T) (𝟭 T) 
+#align category_theory.arrow CategoryTheory.Arrow
+
+/- Porting note: could not derive `Category` above so this instance works in its place-/
+instance : Category (Arrow T) := commaCategory 
+
+-- Satisfying the inhabited linter
+instance Arrow.inhabited [Inhabited T] : Inhabited (Arrow T)
+    where default := show Comma (𝟭 T) (𝟭 T) from default
+#align category_theory.arrow.inhabited CategoryTheory.Arrow.inhabited
+
+end
+
+namespace Arrow
+
+@[simp]
+theorem id_left (f : Arrow T) : CommaMorphism.left (𝟙 f) = 𝟙 f.left :=
+  rfl
+#align category_theory.arrow.id_left CategoryTheory.Arrow.id_left
+
+@[simp]
+theorem id_right (f : Arrow T) : CommaMorphism.right (𝟙 f) = 𝟙 f.right :=
+  rfl
+#align category_theory.arrow.id_right CategoryTheory.Arrow.id_right
+
+/-- An object in the arrow category is simply a morphism in `T`. -/
+@[simps]
+def mk {X Y : T} (f : X ⟶ Y) : Arrow T where
+  left := X
+  right := Y
+  hom := f
+#align category_theory.arrow.mk CategoryTheory.Arrow.mk
+
+@[simp]
+theorem mk_eq (f : Arrow T) : Arrow.mk f.hom = f := by
+  cases f
+  rfl
+#align category_theory.arrow.mk_eq CategoryTheory.Arrow.mk_eq
+
+theorem mk_injective (A B : T) : Function.Injective (Arrow.mk : (A ⟶ B) → Arrow T) := fun f g h =>
+  by
+  cases h
+  rfl
+#align category_theory.arrow.mk_injective CategoryTheory.Arrow.mk_injective
+
+theorem mk_inj (A B : T) {f g : A ⟶ B} : Arrow.mk f = Arrow.mk g ↔ f = g :=
+  (mk_injective A B).eq_iff
+#align category_theory.arrow.mk_inj CategoryTheory.Arrow.mk_inj
+
+/- Porting note : was marked as dangerous instance so changed from `Coe` to `CoeOut` -/
+instance {X Y : T} : CoeOut (X ⟶ Y) (Arrow T) where 
+  coe := mk 
+
+/-- A morphism in the arrow category is a commutative square connecting two objects of the arrow
+    category. -/
+@[simps]
+def homMk {f g : Arrow T} {u : f.left ⟶ g.left} {v : f.right ⟶ g.right}
+    (w : u ≫ g.hom = f.hom ≫ v) : f ⟶ g where
+  left := u
+  right := v
+  w := w
+#align category_theory.arrow.hom_mk CategoryTheory.Arrow.homMk
+
+/-- We can also build a morphism in the arrow category out of any commutative square in `T`. -/
+@[simps]
+def homMk' {X Y : T} {f : X ⟶ Y} {P Q : T} {g : P ⟶ Q} {u : X ⟶ P} {v : Y ⟶ Q} (w : u ≫ g = f ≫ v) :
+    Arrow.mk f ⟶ Arrow.mk g where
+  left := u
+  right := v
+  w := w
+#align category_theory.arrow.hom_mk' CategoryTheory.Arrow.homMk'
+
+/- Porting note : removred simp from reassoc -/
```

Comment:
```suggestion
/- Porting note : removed simp from reassoc -/
```
It looks like the `simp` is still there... is the note out of date?

Reply:
Yes. Corrected it.